### PR TITLE
Add timeout, fix FakeSession (dry run)

### PR DESCRIPTION
**Pull request by mgillis**

### CODE REVIEW

File: s4/clarity/lims.py [hunk 1]
```diff
@@ -259,7 +262,7 @@ def raw_request(self, method, uri, **kwargs):
                 s4.clarity.utils.ssh.tunnel(real_host, 9080, "glsai")
                 self._opened_ssh_tunnel = True
 
-        response = self._session.request(method, uri, allow_redirects=False, **kwargs)
+        response = self._session.request(method, uri, timeout=self.timeout, allow_redirects=False, **kwargs)
```
Comment (Mark-Swinkels) [inline s4/clarity/lims.py:265]: This is going to break a lot of code, we have a lots of clarity calls that take multiple minutes.

**Comment by mgillis** [comment]
Ok, what should the default timeout value be, ten minutes?

**Comment by Mark-Swinkels** [comment]
The default should be None rather than any specific time.

-- Mark

[image: Logo] Mark Swinkels CTO
mark.swinkels@semaphoresolutions.ca
1 250 896 6538
1 844 744 3577
semaphoresolutions.com <http://www.semaphoresolutions.com>
This message and any attachments are confidential and intended for the
named addressee(s) only. If you have received this message in error, please
notify immediately the sender, then delete the message. Any unauthorized
modification, edition, use or dissemination is prohibited. The sender shall
not be liable for this message if it has been modified, altered, falsified,
infected by a virus or even edited or disseminated without authorization.

**Comment by mgillis** [comment]
@Mark-Swinkels resolved

File: s4/clarity/lims.py [hunk 2]
```diff
@@ -26,9 +27,9 @@ class LIMS(object):
     :param str root_uri: Location of the clarity server e.g. (https://<clarity server>/api/v2/)
     :param str username: Clarity User Name
     :param str password: Clarity Password
-    :param bool dry_run: If True, do not actually make calls to Clarity
-    :param bool insecure: Disables SSL validation
-
+    :param bool dry_run: If true, no destructive requests will be made to the Clarity API. Default false.
+    :param bool insecure: Disables SSL validation. Default false.
+    :param int timeout: Number of seconds to wait for connections and for reads from the Clarity API. Default 90s.
```
Comment (Mark-Swinkels) [inline s4/clarity/lims.py:32]: Comment incorrect

**Pull request merged**